File: src/main/java/org/homework20241215/house/HouseBuilder.java
package org.homework20241215.house;

public class HouseBuilder {

    private String address;
    private Integer floors;
    private String floorMaterial;
    private String wallMaterial;

    public HouseBuilder setAddress(String address) {
        this.address = address;
        return this;
    }

    public HouseBuilder setFloors(Integer floors) {
        this.floors = floors;
        return this;
    }

    public HouseBuilder setFloorMaterial(String floorMaterial) {
        this.floorMaterial = floorMaterial;
        return this;
    }

    public HouseBuilder setWallMaterial(String wallMaterial) {
        this.wallMaterial = wallMaterial;
        return this;
    }

    public House build() {
        if (address == null) {
            throw new IllegalStateException("addressが未設定です");
        }
        return new House(address, floors, floorMaterial, wallMaterial);
    }
}

File: src/main/java/org/homework20241222/house/HouseBuilder.kt
package org.homework20241222.house

class HouseBuilder {

    private var address: String ?= null
    // Integer宣言できない
    private var floors: Int ?=null
    private var floorMaterial: String ?= null
    private var wallMaterial: String ?= null

    fun setAddress(address:String) :HouseBuilder {
        this.address = address;
        return this;
    }

    fun setFloors(floors:Int) :HouseBuilder {
        this.floors = floors
        return this
    }

    fun setFloorMaterial(floorMaterial:String) :HouseBuilder {
        this.floorMaterial = floorMaterial
        return this
    }

    fun setWallMaterial(wallMaterial:String):HouseBuilder {
        this.wallMaterial = wallMaterial
        return this
    }

    fun build() :House{
        if (address == null) {
            throw IllegalStateException("addressが未設定です");
        }
        //なぜか「.toString」と書かないとエラーになる
        return House(address.toString(), floors.hashCode(), floorMaterial.toString(), wallMaterial.toString())
    }
}